### PR TITLE
first pass at groups & access docs

### PR DESCRIPTION
**Pull request by MichelleArk**
## What are you changing in this pull request and why?

Addresses: https://github.com/dbt-labs/docs.getdbt.com/issues/2938

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [x] Add page to `website/sidebars.js`
- [x] Provide a unique filename for the new page

### CODE REVIEW

File: website/docs/reference/artifacts/manifest-json.md [hunk 1]
```diff
@@ -46,10 +46,12 @@ Today, dbt uses this file to populate the [docs site](documentation), and to per
 - `sources`: Dictionary of sources.
 - `metrics`: Dictionary of metrics.
 - `exposures`: Dictionary of exposures.
+- `groups`: Dictionary of groups.
 - `macros`: Dictionary of macros.
 - `docs`: Dictionary of `docs` blocks.
 - `parent_map`: Dictionary that contains the first-order parents of each resource.
 - `child_map`: Dictionary that contains the first-order children of each resource.
+- `group_map`: Dictionary that maps group names to their resource nodes.
```
Comment (jtcohen6) [inline website/docs/reference/artifacts/manifest-json.md:54]: Oh, I didn't know we added this - cool!

File: website/docs/reference/dbt-jinja-functions/graph.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ to understand how to effectively use this variable.
 
 <Changelog>
 
+  - In dbt v1.5.0, groups were added to the `graph.groups` object
```
Comment (jtcohen6) [inline website/docs/reference/dbt-jinja-functions/graph.md]: This `<Changelog>` component is legacy, we haven't really used it since v1.0, in favor of new `VersionBlock` components

That does make me realize... do we need to `VersionBlock` the code snippet below, so that only v1.5+ see that it includes `groups`? It's a small hassle, and I don't feel super strongly. @matthewshaver I'm happy to have you make the call.

File: website/docs/reference/dbt-jinja-functions/graph.md [hunk 3]
```diff
@@ -236,3 +248,26 @@ Example usage:
...
+
+To access the groups in your dbt project programmatically, use the `groups` attribute of the `graph` object.
+
+Example usage:
+
+<File name='macros/get_group.sql'>
+
+```sql
+
+{% macro get_group_owner_for(group_name) %}
```
Comment (jtcohen6) [inline website/docs/reference/dbt-jinja-functions/graph.md:255]: Nice toy example!

File: website/docs/reference/resource-properties/access.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+---
+resource_types: [models]
+datatype: access
+required: no
+---
+
+:::info Beta functionality
+This functionality is new in v1.5. These docs exist to provide a high-level overview of what's to come. The specific syntax is liable to change.
```
Comment (jtcohen6) [inline website/docs/reference/resource-properties/access.md]: Given that we're nearing the final beta / RC, I've started using this language (in #3100):
```suggestion
This functionality is new in v1.5! The syntax is mostly locked, but some small details are still liable to change.
```

File: website/sidebars.js [hunk 5]
```diff
@@ -303,49 +304,49 @@ const sidebarSettings = {
...
+          label: "Metadata API",
+          items: [
+            "docs/dbt-cloud-apis/metadata-api",
+            "docs/dbt-cloud-apis/metadata-use-case-guides",
+            "docs/dbt-cloud-apis/access-metadata-api",
+            "docs/dbt-cloud-apis/metadata-querying",
+            {
               type: "category",
-              label: "Metadata API",
+              label: "Schema",
```
Comment (jtcohen6) [inline website/sidebars.js:327]: Looks like an unrelated change from https://github.com/dbt-labs/docs.getdbt.com/pull/3006 that's appearing as a diff here. Not sure why, but this matches the code in the `current` branch

**Review by jtcohen6 — COMMENTED** [review]
LGTM from a technical/content perspective!

@matthewshaver Tagging you in for final editorial review - feel free to propose (or directly make) any changes as you sees fit, and merge to get this live :)

File: website/docs/docs/build/groups.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+title: "Groups"
+id: "groups"
+description: "When you define groups in dbt projects, you turn implicit relationships into an explicit grouping."
+keywords:
+  - groups access mesh
+---
+
+## About Groups 
+
+A group is collection of resources within a dbt project. Groups are named, and every group has an `owner`. They enable intentional collaboration within and across teams, by restricting [access to private](access) models
```
Comment (matthewshaver) [inline website/docs/docs/build/groups.md]:
```suggestion
A group is a collection of resources within a dbt project. Groups are named, and every group has an `owner`. They enable intentional collaboration within and across teams by restricting [access to private](access) models
```

File: website/docs/docs/build/groups.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+title: "Groups"
+id: "groups"
+description: "When you define groups in dbt projects, you turn implicit relationships into an explicit grouping."
+keywords:
+  - groups access mesh
+---
+
+## About Groups 
+
+A group is a collection of resources within a dbt project. Groups are named, and every group has an `owner`. They enable intentional collaboration within and across teams by restricting [access to private](access) models
```
Comment (matthewshaver) [inline website/docs/docs/build/groups.md]:
```suggestion
A group is a collection of resources within a dbt project. Groups are named, and every group has an `owner`. They enable intentional collaboration within and across teams by restricting [access to private](access) models.
```

File: website/docs/reference/dbt-jinja-functions/graph.md (hunk 2 above)
Comment (matthewshaver) [reply website/docs/reference/dbt-jinja-functions/graph.md]: Let's version blocks here only because I want to keep things consistent across the platform and reduce any future tech debt.

File: website/docs/reference/resource-configs/group.md [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+```
+
+</File>
+
+</TabItem>
+
+</Tabs>
+
+## Definition
+An optional configuration for grouping models, analysis, snapshots, tests and metrics. When a resource is grouped, dbt will allow it to reference private models within the same group.
```
Comment (matthewshaver) [inline website/docs/reference/resource-configs/group.md]:
```suggestion
An optional configuration for grouping models, analysis, snapshots, tests, and metrics. When a resource is grouped, dbt will allow it to reference private models within the same group.
```

File: website/docs/docs/build/groups.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+models:
+  - name: model_name
+    group: finance
+```
+
+</File>
+
+### Referencing a model in a group
+
+By default, all models within a group are `protected` in access - meaning they can be referenced by downstream resources in _any_ group in the project using the [`ref`](ref) function. If a grouped model's `access` property is set to `private`, only resources within its group will be able to make references to it.
```
Comment (matthewshaver) [inline website/docs/docs/build/groups.md]:
```suggestion
By default, all models within a group are `protected` in access - meaning they can be referenced by downstream resources in _any_ group in the project using the [`ref`](ref) function. If a grouped model's `access` property is set to `private`, only resources within its group can reference it. 
```

File: website/docs/reference/resource-configs/group.md [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+</Tabs>
+
+## Definition
+An optional configuration for grouping models, analysis, snapshots, tests, and metrics. When a resource is grouped, dbt will allow it to reference private models within the same group.
+
+For more details on reference access between resources in groups, check out [model access](/docs/collaborate/publish/model-access#groups).
+
+## Examples
+### Prevent a 'marketing' group model from referencing a private 'finance' group model
+This could be useful if you want to prevent other groups from building on top on models that are rapidly changing, experimental, or otherwise internal to a group or team.
```
Comment (matthewshaver) [inline website/docs/reference/resource-configs/group.md]:
```suggestion
This is useful if you want to prevent other groups from building on top of models that are rapidly changing, experimental, or otherwise internal to a group or team. 
```

**Review by matthewshaver — APPROVED** [review]
